src/Adapter/Order/CommandHandler/ChangeOrderDeliveryAddressHandler.php

```diff
@@ -51,16 +61,32 @@ final class ChangeOrderDeliveryAddressHandler extends AbstractOrderHandler imple
      */
     private $orderDetailTaxUpdater;
 
+    /**
+     * @var ContextStateManager
+     */
+    private $contextStateManager;
+
+    /**
+     * @var OrderProductQuantityUpdater
+     */
+    private $orderProductQuantityUpdater;
+
     /**
      * @param OrderAmountUpdater $orderAmountUpdater
      * @param OrderDetailUpdater $orderDetailTaxUpdater
+     * @param ContextStateManager $contextStateManager
```

Comment:
I don't think we should do the same thing for `ChangeOrderInvoiceAddressHandler` as we do not recalculate cart rules when changing the invoice address

Reply:
We do recompute the cart rules amounts in case the address changes its country and tax are related to the invoice address But it shouldn't impact the gifts indeed And the context state manager is already handled internally by `orderAmountUpdater` and `orderDetailTaxUpdater`
So you're right it should be good